cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -89,6 +171,26 @@ private module Cached {
   Expr getInstructionUnconvertedResultExpression(Instruction instruction) {
     result = getInstructionConvertedResultExpression(instruction).getUnconverted()
   }
+}
+
+import Cached
+
+cached
+private module Cached {
+  class TStageInstruction = TRawInstruction;
+
+  cached
+  predicate hasInstruction(TRawInstruction instr) { any() }
```

Comment:
Can this get optimised away if it's moved out of `Cached`?

Reply:
I tried that, and while it gets optimized away, another uncached predicate (`TInstruction::TRawInstruction#ffffff_5#antijoin_rhs`) with the same implementation gets created instead for some other use case.